### PR TITLE
ci: remove ci workspace

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
We no longer have prebuilt dependencies, so ci/WORKSPACE is no longer needed. Simplifies how CI runs.

Risk Level: Med for depending projects
Testing: CI
Docs Changes:
Release Notes:

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -268,7 +268,7 @@ elif [[ "$1" == "bazel.coverage" ]]; then
   exit 0
 elif [[ "$1" == "bazel.clang_tidy" ]]; then
   setup_clang_toolchain
-  cd "${ENVOY_CI_DIR}"
+  cd "${ENVOY_SRCDIR}"
```
Comment (htuch) [inline ci/do_ci.sh]: Aren't we already in ENVOY_SRCDIR in all ofthese?

**Comment by lizan** [comment]
Yeah I might have to roll #6186 into this PR to make it pass all CI runs.

**Comment by wu-bin** [comment]
@lizan Any update on PR?  I think this will allow us to [get rid of envoy_select_quiche](https://github.com/envoyproxy/envoy/pull/6393)

**Pull request merged**